components/org.wso2.carbon.identity.application.authenticator.fido2/src/main/java/org/wso2/carbon/identity/application/authenticator/fido2/internal/MetadataService.java

```diff
@@ -0,0 +1,180 @@
+/*
+ * Copyright (c) 2022, WSO2 Inc. (http://www.wso2.com).
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.application.authenticator.fido2.internal;
+
+import com.webauthn4j.anchor.TrustAnchorRepository;
+import com.webauthn4j.converter.util.ObjectConverter;
+import com.webauthn4j.metadata.FidoMDS3MetadataBLOBProvider;
+import com.webauthn4j.metadata.LocalFilesMetadataStatementsProvider;
+import com.webauthn4j.metadata.MetadataBLOBProvider;
+import com.webauthn4j.metadata.anchor.AggregatingTrustAnchorRepository;
+import com.webauthn4j.metadata.anchor.MetadataBLOBBasedTrustAnchorRepository;
+import com.webauthn4j.metadata.anchor.MetadataStatementsBasedTrustAnchorRepository;
+import com.webauthn4j.validator.attestation.trustworthiness.certpath.DefaultCertPathTrustworthinessValidator;
+import org.apache.commons.lang.StringUtils;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.carbon.identity.application.authenticator.fido2.util.FIDO2AuthenticatorConstants;
+import org.wso2.carbon.identity.core.util.IdentityConfigParser;
+import org.wso2.carbon.identity.core.util.IdentityUtil;
+
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.security.cert.CertificateException;
+import java.security.cert.CertificateFactory;
+import java.security.cert.X509Certificate;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Objects;
+
+/**
+ * Helper class for FIDO2 metadata validations.
+ */
+public class MetadataService {
+
+    private static final Log log = LogFactory.getLog(MetadataService.class);
+    private ObjectConverter objectConverter = null;
+    private DefaultCertPathTrustworthinessValidator defaultCertPathTrustworthinessValidator = null;
+    private static ArrayList<String> mdsEndpoints = null;
+
+    public MetadataService() {
+    }
+
+    /**
+     * Initialize the DefaultCertPathTrustworthinessValidator object needed for webauthn4j mds validations.
+     */
+    public void initializeDefaultCertPathTrustworthinessValidator() {
+
+        objectConverter = new ObjectConverter();
+
```

Comment:
call `getMDS3RootCertificate()` somewhere here

Reply:
Fixed.
Will throw an exception if the root certificate format is invalid or the file is not found.